### PR TITLE
Explore express migration + Add auth guard

### LINKED ISSUE #38: Add Auth Guard to GraphQL

### PR DESCRIPTION
**Pull request by norisalsaadie**
Closes #38 but also moves everything away from nest

### CODE REVIEW

File: src/repositories/submission.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { SubmissionRepository, DtoSubmission, SubmissionId, Submission } from '../types/submission';
+import * as Knex from 'knex';
+import { InfraLogger as logger } from '../logger';
+import { SubmissionMapper } from '../entities/submission';
+
+export class KnexSubmissionRepository implements SubmissionRepository {
+    private readonly TABLE_NAME = 'submission';
+    // private readonly logger = new Logger(KnexSubmissionRepository.name);
```
Comment (diversemix) [inline src/repositories/submission.ts]: this need going back in?

Comment (norisalsaadie) [reply src/repositories/submission.ts]: The logger is now the same as https://github.com/libero/continuum-adaptor/blob/master/src/logger.ts

Comment (norisalsaadie) [reply src/repositories/submission.ts]: That being said, schema innit logic should probably be remove. .. https://github.com/libero/continuum-adaptor/blob/master/src/logger.ts

It should be part of a migration... but this PR is already large enough....

File: README.md [hunk 2]
```diff
@@ -1,12 +1,27 @@
 # Reviewer Submission
 
+## Design and folder structure
+
+The application uses the following:
+  * TypeScript
+  * Express.js
+  * GraphQL (Apollo Server)
+
+The structre is as follows:
```
Comment (diversemix) [inline README.md]: :+1:

File: src/repositories/submission.ts (hunk 1 above)
Comment (diversemix) [reply src/repositories/submission.ts]: ok - we should remove this line then?

File: package.json [hunk 3]
```diff
@@ -63,23 +66,25 @@
...
-        "@types/express": "^4.16.0",
+        "@types/express": "^4.17.2",
+        "@types/helmet": "^0.0.45",
         "@types/jest": "^23.3.13",
+        "@types/jsonwebtoken": "^8.3.7",
+        "@types/knex": "^0.16.1",
         "@types/lodash": "^4.14.149",
         "@types/node": "^10.12.18",
         "@types/node-fetch": "^2.5.4",
         "@types/supertest": "^2.0.7",
```
Comment (Cooryd) [inline package.json]: Wondering if its worth auditing the dependencies to see if any are unused now? Packages like `sqlite3` might be removable

**Pull request merged**